experimental/oak_baremetal_channel/src/lib.rs

```diff
@@ -88,6 +96,7 @@ const MAX_FRAME_BODY_SIZE: usize = MAX_FRAME_SIZE - FRAME_HEADER_SIZE;
 #[derive(Clone)]
 struct Frame {
     method_or_status: MethodOrStatus,
+    message_id: MessageId,
     flag: Flag,
```

Comment:
incidentally, I feel more and more that this should probably be turned into a bitmap of flags, instead of a single flag field.

Reply:
+1: either that, or rename the field. The expection is usually that something named `flag` is a boolean and something names `flags` represents a set of bit flags.

But that can be included as part of the next framing layer redesign iteration.